In [3]:
import pandas as pd

orders = pd.read_csv("E:/Blinkit_Business_Analytics/Dataset/Clean_Data/orders_clean.csv")
customers = pd.read_csv("E:/Blinkit_Business_Analytics/Dataset/Clean_Data/customers_clean.csv")
products = pd.read_csv("E:/Blinkit_Business_Analytics/Dataset/Clean_Data/products_clean.csv")
delivery = pd.read_csv("E:/Blinkit_Business_Analytics/Dataset/Clean_Data/delivery_clean.csv")
inventory = pd.read_csv("E:/Blinkit_Business_Analytics/Dataset/Clean_Data/inventory_clean.csv")
feedback = pd.read_csv("E:/Blinkit_Business_Analytics/Dataset/Clean_Data/feedback_clean.csv")

In [4]:
orders["order_total"].describe()

count    5000.00000
mean     2201.86170
std      1303.02438
min        13.25000
25%      1086.21500
50%      2100.69000
75%      3156.88250
max      6721.46000
Name: order_total, dtype: float64

In [5]:
orders["payment_method"].value_counts()

payment_method
Card      1285
Cash      1257
Wallet    1244
UPI       1214
Name: count, dtype: int64

In [6]:
orders["delivery_status"].value_counts()

delivery_status
On Time                  3470
Slightly Delayed         1037
Significantly Delayed     493
Name: count, dtype: int64

In [7]:
orders["order_total"].sum()

np.float64(11009308.5)

In [8]:
orders["order_total"].mean()

np.float64(2201.8617)

In [9]:
orders["order_date"] = pd.to_datetime(orders["order_date"])

orders["Month"] = orders["order_date"].dt.month_name()

orders.groupby("Month")["order_total"].sum()

Month
April        1093099.52
August       1169666.92
December      615709.03
February      545090.11
January       560423.56
July         1140751.89
June         1044302.51
March         816060.81
May          1182377.15
November      635941.39
October      1116072.77
September    1089812.84
Name: order_total, dtype: float64

In [10]:
customers["customer_segment"].value_counts()

customer_segment
Regular     639
Premium     633
New         628
Inactive    600
Name: count, dtype: int64

In [11]:
customers["avg_order_value"].describe()

count    2500.000000
mean     1102.375836
std       523.043183
min       200.430000
25%       631.817500
50%      1118.650000
75%      1565.400000
max      1999.830000
Name: avg_order_value, dtype: float64

In [12]:
delivery["delivery_status"].value_counts()

delivery_status
On Time                  3470
Slightly Delayed         1037
Significantly Delayed     493
Name: count, dtype: int64

In [13]:
delivery["delivery_time_minutes"].describe()


count    5000.000000
mean        4.443000
std         8.063929
min        -5.000000
25%        -1.000000
50%         2.000000
75%         8.000000
max        30.000000
Name: delivery_time_minutes, dtype: float64

In [19]:
late_deliveries = delivery[
    delivery["delivery_status"].isin([
        "Slightly Delayed",
        "Significantly Delayed"
    ])
]

late_deliveries.head()

,order_id,delivery_partner_id,promised_time,actual_time,delivery_time_minutes,distance_km,delivery_status,reasons_if_delayed
9,374186990,77203,2023-08-09 01:37:30,2023-08-09 01:44:30,7.0,1.47,Slightly Delayed,Traffic
12,9171116867,73907,2023-10-22 11:25:33,2023-10-22 11:37:33,12.0,2.23,Slightly Delayed,Traffic
13,5504941865,56077,2024-01-28 02:55:11,2024-01-28 03:06:11,11.0,4.45,Slightly Delayed,Traffic
16,8195544338,102,2023-04-26 04:20:57,2023-04-26 04:28:57,8.0,1.16,Slightly Delayed,Traffic
19,4656471694,62782,2023-12-09 08:27:08,2023-12-09 08:35:08,8.0,3.70,Slightly Delayed,Traffic


In [20]:
len(late_deliveries)

1530

In [14]:
feedback["rating"].mean()

np.float64(3.3444)

In [15]:
feedback["sentiment"].value_counts()

sentiment
Neutral     1738
Negative    1642
Positive    1620
Name: count, dtype: int64

In [16]:
feedback["feedback_category"].value_counts()

feedback_category
Delivery            1271
Customer Service    1266
Product Quality     1250
App Experience      1213
Name: count, dtype: int64

In [17]:
inventory["stock_received"].sum()

np.int64(147526)

In [18]:
inventory["damaged_stock"].sum()

np.int64(80268)

Business Insights

1. Total Revenue = 11009308.5
2. Average Order Value = 2201.8617
3. Most used payment method = card
4. Most common customer segment = Regular
5. Average rating = 3.3444
6. Late deliveries = 1530
7. Total damaged stock =  802668 units